### ISSUE TITLE
Some places shows {0} placeholder instead of expected value - Portuguese

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#16690]
**Describe the bug**
Reprise of #15694, but only for instances where language was defined as "Portuguese" prior to pre-39.0 and then upgrading to 0.39.4.
Pre-39 had "Portuguese" as `pt` internally, but that was changed in 0.39.0 (#15630) to "Portuguese (Brazil)" `pt-BR`.

Though, I don't understand what changed in 0.39.4 that suddenly caused this problem to occur, when it works on 0.39.3.

**Workaround** Go and set the language to "Portuguese (Brazil)" again in Admin > Settings > Localization. Note that it will display "en" after setting the language, but that's issue #15657.

**To Reproduce**
1. Admin > Settings > Localization > (set language value to `pt` via the API)
Note: Very likely the same problem for user-defined language under their Account settings, but haven't tested.
2. Admin > Settings > Updates - or - Simple question > Sample Dataset > Orders - will show `{0}` placeholders
Note: There are many other places that will show placeholders like that.
![image](https://user-images.githubusercontent.com/1447303/122648068-9e9dbf00-d127-11eb-80a4-045064d1355d.png)
![image](https://user-images.githubusercontent.com/1447303/122648006-67c7a900-d127-11eb-967e-656f7e9e74e9.png)

**Information about your Metabase Installation:**
Regression since 0.39.4, but seeded by a change in 0.39.0

**Additional context**
Related #15657/#16654
Re-write of #16681

**Comment by kulyk** [comment]
Hey @flamber, can you still reproduce the issue? Seems fine for me locally on the latest `master`

### Admin

![image](https://user-images.githubusercontent.com/17258145/127166725-d7021fc9-510b-42c3-9e2a-ee04288ad286.png)

### Question bottom bar

![image](https://user-images.githubusercontent.com/17258145/127166830-1ec51b1c-6579-433b-9af2-de8193cac1b8.png)

**Comment by flamber** [comment]
@kulyk We've had several reports of this the past couple of weeks, when people were upgrading to 40.0 and 40.1.
This issue is specific. You need to set the language on 38, then upgrade to 39.4+ to trigger the problem.
Try reproducing that way instead of through API, which might do something different now, so it cannot be reproduced that way.

**Comment by kulyk** [comment]
Wow, gotcha. Thank you, will investigate

**Comment by pawit-metabase** [comment]
On `master`, if the language is `pt` in the database, the UI now just displays in English.

Should we just have a migration that change `pt` to `pt-BR` for site setting and all users?

**Comment by flamber** [comment]
@pawit-metabase Probably, since it doesn't seem like there will be a plain Portuguese translation [anytime soon](https://poeditor.com/join/project/ynjQmwSsGh) (1% translated)
But if plain Portuguese is added in later releases, then the migration has already been executed, so that would not conflict.

Though, migrations are a lot more difficult, since it doesn't change encrypted site settings (table `setting`).

**Comment by pawit-metabase** [comment]
Oh, then database migration is not going to work. If we want smooth migration, then we'd have to change it when the code load it from the DB and then remove the code later when we actually have `pt` translation.

**Comment by flamber** [comment]
I think that might be the best option right now, if it is easy to do in code.

**Comment by pawit-metabase** [comment]
Another observation - the release jar contains old locale from unclean build directory. This file shouldn't exist since it was renamed to pt-BR/pt_BR. This is likely the reason it happens on release but cannot be reproduced locally.

```
zipinfo metabase.jar | grep -E 'pt\.(json|edn)'
-rw----     2.0 fat   216674 bl defN 21-Jun-15 13:14 frontend_client/app/locales/pt.json
-rw----     2.0 fat   126820 bl defN 21-Jun-15 13:14 i18n/pt.edn
```

**Comment by flamber** [comment]
That makes sense why it suddenly appeared in 0.39.4

**Comment by flamber** [comment]
This is automatically "fixed" with a clean build environment in 0.40.5, but #17875 should fully prevent this in the future.
